frontend/src/features/user/transfer-ownership/TransferOwnershipModal.tsx

```diff
@@ -0,0 +1,189 @@
+import { useCallback, useEffect, useState } from 'react'
+import { SubmitHandler, useForm } from 'react-hook-form'
+import {
+  Button,
+  ButtonGroup,
+  FormControl,
+  FormErrorMessage,
+  FormLabel,
+  Input,
+  Modal,
+  ModalBody,
+  ModalContent,
+  ModalHeader,
+  ModalOverlay,
+  Text,
+  useBreakpointValue,
+} from '@chakra-ui/react'
+import isEmail from 'validator/lib/isEmail'
+
+import {
+  CANNOT_TRANSFER_OWNERSHIP_TO_SELF,
+  INVALID_EMAIL_ERROR,
+  REQUIRED_ERROR,
+} from '~constants/validation'
+import { ModalCloseButton } from '~components/Modal'
+
+import { useUserMutations } from '~features/user/mutations'
+
+import { useUser } from '../queries'
+
+interface TransferOwnershipModalProps {
+  isOpen: boolean
+  onClose: () => void
+}
+
+type TransferOwnershipInputs = {
+  email: string
+}
+
+const useModalState = ({ onClose, reset, trigger }) => {
+  const [page, setPage] = useState(0)
+  const [email, setEmail] = useState('')
+
+  const { user } = useUser()
+  const { transferOwnershipMutation } = useUserMutations()
+
+  const isOwnEmail = useCallback(
+    (value: string) => {
+      return user?.email && value.toLowerCase() === user.email
+    },
+    [user?.email],
+  )
+
+  const resetModal = useCallback(() => {
+    setPage(0)
+    reset()
+    trigger()
+    onClose()
+  }, [page, email])
+
+  const onNext: SubmitHandler<TransferOwnershipInputs> = useCallback(
+    ({ email }) => {
+      setEmail(email)
+      setPage(1)
+    },
+    [page, email],
+  )
+
+  const onConfirm = useCallback(() => {
+    if (!user) return
+    return transferOwnershipMutation.mutate(
+      { email },
+      {
+        onSuccess: () => {
+          resetModal()
+        },
```

Comment:
How about for failure case - should we show a toast and ask the user to retry?

Reply:
Correct me if I'm wrong: I thought the `mutations.ts` file already has a toast (also left [a comment](https://github.com/opengovsg/FormSG/pull/6087/files#r1278838934) there).

But the content (i.e. `error.message`) does not specifically ask the user to retry. Do you want me to do something like:

```
`${error.message}: Please try again.`
```

Or do you also want me to move out the toast call into `TransferOwnershipModal.tsx`?

Reply:
Done!